ada/language/ast.py

```diff
@@ -1037,6 +1042,45 @@ def env_get_real_from_node(from_node=T.AdaNode):
             ): If(c.is_null, from_node, c)
         ))
 
+    @langkit_property(public=True, return_type=T.AdaNode.entity.array,
+                      dynamic_vars=[origin, default_imprecise_fallback()])
+    def find_all_in(x=T.AdaNode.entity, mode=FindAllMode):
```

Comment:
Please name `x` `root`, will be a bit clearer. Also `find_all_in` could be named `find_all_refs_in`, feels like you a word.

Reply:
Ok for renaming `X` to `Root`, however, I renamed `find_all_refs_in` to `find_all_in` on purpose, because this is not anymore a procedure for finding _references_. It's just a tree traversal that you can use to collect whatever kind of node you need (references, accept statements, etc.)